usmqe_tests/api/tendrl/test_volume.py

```diff
@@ -0,0 +1,280 @@
+            ``gluster volume info command``
+
+        .. test_result:: 2
+
+            There should be listed gluster volume named ``Vol_test``.
+
+            """
+    storage = gluster.GlusterCommon()
+    storage.find_volume_name(valid_volume_name)
+
+    # TODO change
+    xml = storage.run_on_node(
+        command="volume info {}".format(valid_volume_name))
+    volume_id = xml.findtext("./volInfo/volumes/volume/id")
+    name = api.get_volume_list(valid_cluster_id)[0][volume_id]["name"]
+    pytest.check(name == valid_volume_name)
+
+
+# TODO create negative test case generator
+# http://doc.pytest.org/en/latest/parametrize.html#basic-pytest-generate-tests-example
+def test_stop_volume_invalid(valid_cluster_id, invalid_volume_name):
+    api = tendrlapi.ApiGluster()
+    volume_data = {
+        "Volume.volname": invalid_volume_name,
+    }
+
+    job_id = api.stop_volume(valid_cluster_id, volume_data)["job_id"]
+    # TODO check correctly server response or etcd job status
+    api.wait_for_job_status(
+            job_id,
+            status="failed",
+            issue="https://github.com/Tendrl/tendrl-api/issues/33")
+
+
+def test_stop_volume_valid(valid_cluster_id, valid_volume_name, valid_volume_id):
+    api = tendrlapi.ApiGluster()
+    volume_data = {
+        "Volume.volname": valid_volume_name,
+    }
+
+    job_id = api.stop_volume(valid_cluster_id, volume_data)["job_id"]
+    api.wait_for_job_status(job_id)
+    test_gluster = gluster.GlusterCommon()
+    test_gluster.check_status(valid_volume_name, "Stopped")
+    status = api.get_volume_list(valid_cluster_id)[0][valid_volume_id]["status"]
+    pytest.check(status == "Stopped", issue="https://github.com/Tendrl/tendrl-api/issues/56")
+
+
+# TODO create negative test case generator
+# http://doc.pytest.org/en/latest/parametrize.html#basic-pytest-generate-tests-example
+def test_start_volume_invalid(valid_cluster_id, invalid_volume_name):
+    api = tendrlapi.ApiGluster()
+    volume_data = {
+        "Volume.volname": invalid_volume_name
+    }
+
+    job_id = api.start_volume(valid_cluster_id, volume_data)["job_id"]
+    # TODO check correctly server response or etcd job status
+    api.wait_for_job_status(
+            job_id,
+            status="failed",
+            issue="https://github.com/Tendrl/tendrl-api/issues/33")
+
+
+def test_start_volume_valid(valid_cluster_id, valid_volume_name, valid_volume_id):
+    api = tendrlapi.ApiGluster()
+    volume_data = {
+        "Volume.volname": valid_volume_name,
+    }
+
+    job_id = api.start_volume(valid_cluster_id, volume_data)["job_id"]
+    api.wait_for_job_status(job_id)
+    test_gluster = gluster.GlusterCommon()
+    test_gluster.check_status(valid_volume_name, "Started")
+    status = api.get_volume_list(valid_cluster_id)[0][valid_volume_id]["status"]
+    pytest.check(status == "Started", issue="https://github.com/Tendrl/tendrl-api/issues/55")
+
+
+# TODO create negative test case generator
```

Comment:
For this particular case I don't think a generator is needed, because it has only one parameter